src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -1569,6 +1570,11 @@ private void SkipChunkDataAndCrc(in PngChunk chunk)
     [MethodImpl(InliningOptions.ShortMethod)]
     private IMemoryOwner<byte> ReadChunkData(int length)
     {
+        if (length == 0)
```

Comment:
We avoid the read here because Png still has to do a CRC check which if does not occur leads to failure during reading of chained Png images within a single stream.